packages/interactivity/src/proxies/test/deep-merge.ts

```diff
@@ -462,5 +465,61 @@ describe( 'Interactivity API', () => {
 			expect( target.message.content ).toBeUndefined();
 			expect( target.message.fontStyle ).toBeUndefined();
 		} );
+
+		it( 'should keep reactivity of objects that are initially undefined', () => {
+			const target: any = proxifyState( 'test', {} );
+
+			let deepValue: any;
+			const spy = jest.fn( () => {
+				deepValue = target.obj?.nested;
+			} );
+			effect( spy );
+
+			// Initial call, the deep value is undefined
+			expect( spy ).toHaveBeenCalledTimes( 1 );
+			expect( deepValue ).toBeUndefined();
+
+			// Use deepMerge to add a deeply nested object to the target
+			deepMerge( target, { obj: { nested: 'value 1' } } );
+
+			// The effect should be called again
+			expect( spy ).toHaveBeenCalledTimes( 2 );
+			expect( deepValue ).toBe( 'value 1' );
+
+			// Modify the nested value
+			target.obj.nested = 'value 2';
+
+			// The effect should be called again
+			expect( spy ).toHaveBeenCalledTimes( 3 );
+			expect( deepValue ).toBe( 'value 2' );
+		} );
+
```

Comment:
This test is essentially the same as:

https://github.com/WordPress/gutenberg/blob/bf3893033b878d17fad6a55c17b91e27042504bf/packages/interactivity/src/proxies/test/deep-merge.ts#L396-L418

with just one extra step. Should we instead modify that existing test?

Reply:
Yes, it makes total sense. Thanks @michalczaplinski. 😄